### PR TITLE
I2C updates

### PR DESCRIPTION
**Pull request by thejpster**
Looks like I lost the rights to push to my old repo when I moved it to community.

### CODE REVIEW

File: tm4c-hal/src/i2c.rs [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+                T: OutputMode,
+            {}
+        )*
+    }
+}
+
+#[macro_export]
+/// Spins on a given field on a TM4C I2C peripheral
+macro_rules! i2c_busy_wait {
+    ($i2c:expr) => {
```
Comment (dtwood) [inline tm4c-hal/src/i2c.rs:49]:
```suggestion
    ($i2c:expr) => {{
```

File: tm4c-hal/src/i2c.rs [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        } else if mcs.arblst().bit_is_set() {
+            Err(Error::Arbitration)
+        } else if mcs.datack().bit_is_set() {
+            Err(Error::DataAck)
+        } else if mcs.adrack().bit_is_set() {
+            Err(Error::AdrAck)
+        } else {
+            Ok(())
+        }
+    };
```
Comment (dtwood) [inline tm4c-hal/src/i2c.rs:81]:
```suggestion
    }};
```

**Comment by dtwood** [comment]
> Looks like I lost the rights to push to my old repo when I moved it to community.

Weird, as I apparently can push there

File: tm4c-hal/src/i2c.rs (hunk 1 above)
Comment (thejpster) [reply tm4c-hal/src/i2c.rs:49]: Huh. Cargo fmt took those out...

**Comment by dkm** [comment]
Great to see this, as I'm seeing random i2c errors when sending a lot of requests to a simple RTC. I will see if this is fixing my problem, at least :)

**Pull request merged**